middle_end/flambda/from_lambda/closure_conversion.ml

```diff
@@ -1037,27 +1035,31 @@ and close_one_function t ~external_env ~by_closure_id decl
       ~return_continuation:(Function_decl.return_continuation decl)
       exn_continuation params ~dbg ~body ~my_closure
   in
+  t.code <- (code_id, params_and_body) :: t.code;
   let fun_decl =
-    Flambda.Function_declaration.create ~code_id
-      ~params_arity:(Kinded_parameter.List.arity params)
-      ~result_arity:[LC.value_kind return]
-      ~stub
-      ~dbg
-      ~inline
-      ~is_a_functor:(Function_decl.is_a_functor decl)
-      ~recursive
+    match Function_decl.kind decl with
+    | Curried ->
+      Flambda.Function_declaration.create ~code_id
+        ~params_arity:(Kinded_parameter.List.arity params)
+        ~result_arity:[LC.value_kind return]
+        ~stub
+        ~dbg
+        ~inline
+        ~is_a_functor:(Function_decl.is_a_functor decl)
+        ~recursive
+        ~is_tupled:false
+    | Tupled ->
+      Flambda.Function_declaration.create ~code_id
+        ~params_arity:[K.value]
+        ~result_arity:[LC.value_kind return]
+        ~stub
+        ~dbg
+        ~inline
+        ~is_a_functor:(Function_decl.is_a_functor decl)
+        ~recursive
+        ~is_tupled:true
```

Comment:
This is a bit too similar to duplicate completely.

Reply:
Fixed